packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch

```diff
@@ -0,0 +1,77 @@
+From 97e0ed83b07b46f0487ed35dbc4a3e61ece95a74 Mon Sep 17 00:00:00 2001
+From: Piyush Jena <piyushjena1996@gmail.com>
+Date: Fri, 23 May 2025 21:22:21 +0000
+Subject: [PATCH] mkfs: source defaults from config file
+
+Signed-off-by: Piyush Jena <piyushjena1996@gmail.com>
+---
+ mkfs/xfs_mkfs.c | 35 +++++++++++++++++++++++++++++++++--
+ 1 file changed, 33 insertions(+), 2 deletions(-)
+
+diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
+index 3f4455d4..fce7d831 100644
+--- a/mkfs/xfs_mkfs.c
++++ b/mkfs/xfs_mkfs.c
+@@ -40,6 +40,10 @@
+  */
+ #define WHACK_SIZE (128 * 1024)
+
++/* Default path for the mkfs.xfs configuration file */
++#define DEFAULT_CONFIG_PATH "/usr/share/xfsprogs/mkfs/default.conf"
++static bool				cfgdefs = false;
++
+ /*
+  * XXX: The configured block and sector sizes are defined as global variables so
+  * that they don't need to be passed to getnum/cvtnum().
+@@ -1579,12 +1583,14 @@ check_opt(
+ 	 * conversion into a number and bounds checking. Hence the two variables
+ 	 * used to track the different uses based on the @str parameter passed
+ 	 * to us.
++	 * Added cfgdefs to make sure we don't set the seen value when configuring
++	 * defaults to allow reassignment through command line parameters.
+ 	 */
+-	if (!str_seen) {
++	if (!str_seen && !cfgdefs) {
+ 		if (sp->seen)
+ 			respec(opts->name, opts->subopts, index);
+ 		sp->seen = true;
+-	} else {
++	} else if (!cfgdefs) {
```

Comment:
You may as well put an `if (!cfgdefs) { ... }` block around the existing logic. It all ends up in the context anyhow, so it doesn't make the patch any more or less brittle, and that will clarify the intent that neither of the two branches is taken if `cfgdefs` is set.